modules/acm/main.tf

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cluster_endpoint       = "https://${var.cluster_endpoint}"
```

Comment:
Couldn't this be fetched from `data.google_container_cluster.primary.endpoint` as per the documentation https://www.terraform.io/docs/providers/google/d/google_container_cluster.html ?

Reply:
Depending on a data source for the cluster endpoint breaks the dependency graph if this module is in the same configuration as the cluster. We witnessed the cluster being destroyed before the resources dependent on the endpoint.